src/polygon.jl

```diff
@@ -0,0 +1,508 @@
+    midpoint = (triangle.p1 + triangle.p2 + triangle.p3)/3
+    dist1 = abs(triangle.p1 - midpoint)
+    dist2 = abs(triangle.p2 - midpoint)
+    dist3 = abs(triangle.p3 - midpoint)
+    isStretchyBoi = max(dist1, dist2, dist3) > 2*min(dist1, dist2, dist3) && return true
+    # An elongated triangle has a big difference in max distance from a point to the centroid to min distance from a point to the centroid (2 is a good threshhold)
+
+    return false # Only gets to this return false if all the other returns true coniditions weren't met (and thus the triangle isnt stupid)
+end
+
+"""
+    generatetriangle(m::Int, n::Int)
+
+Generates a random triangle (a non-stupid one at that).
+"""
+function generatetriangle(m::Int, n::Int)
+    badTriangle = true
+    t = nothing
+    while badTriangle
+        center = complex(rand()*1.1*n - 0.05*n, rand()*1.1*m - 0.05*m)
+        deviations = complex.(rand(3)*n*4/5 .- n*2/5, rand(3)*m*4/5 .- m*2/5)
+        points = Complex{Int}.(round.(center .+ deviations))
+        # Points are allowed to be outside of canvas as long as part of the triangle is in the canvas
+        # Prevents white borders
+        col = RGB(rand(), rand(), rand())
+        t = Triangle(points[1], points[2], points[3], col)
+        badTriangle = checktriangle(t, m, n)
+    end
+
+    return t
+end
+
+"""
+    generatepopulation(pop_size::Int, number_triangles::Int, img::Array)
+
+Generates an entire population. The individuals of this population consist of:
+- An array of triangles (remember, a triangle is simply 3 xy coordinates of points and a color, not yet an image)
+- The image made from drawing all these triangles onto a white canvas (array of triangles converted to an image)
+- The score of the individual, which is too say how much its image differs from the target image.
+"""
+function generatepopulation(pop_size::Int, number_triangles::Int, img::Array)
+
+    m = length(img[:, 1])
+    n = length(img[1, :])
+    canvas = fill(RGB(1, 1, 1), m, n)
+
+    population = Vector{Array}(undef, pop_size)
+
+    for i in 1:pop_size
+        triangles = [generatetriangle(m, n) for i in 1:number_triangles]
+        polyimg = drawimage(triangles, canvas)
+
+        score = colordiffsum(polyimg, img)
+        population[i] = [triangles, polyimg, score]
+    end
+    return population
+end
+
+
+"""
+    triangletournament(population0::Array, pop_size::Int)
+
+Picks 2 random triangles from a population and returns the one with the best (= lowest) score.
+"""
+function triangletournament(population0::Array, pop_size::Int)
+    contestant1 = population0[Int(ceil(rand()*pop_size))] #Choose a random image from the population, using ceil because 0 is not a valid index
+    contestant2 = population0[Int(ceil(rand()*pop_size))]
+
+    while contestant1 == contestant2 #If it happened to pick the same individual twice, change it until you have 2 unique individuals
+        contestant2 = population0[Int(ceil(rand()*pop_size))]
+    end
```

Comment:
Maybe it'd be a bit cleaner to exclude the first contestant from the array before choosing the second one?